From 6d4cca9c22c52b5a7742adc47ebf1a2930e29764 Mon Sep 17 00:00:00 2001
From: David Tardon <dtardon@redhat.com>
Date: Wed, 14 Mar 2012 13:58:37 +0100
Subject: [PATCH] fix setting of paper tray from print dialog (fdo#43932)

This fixes the simple case when all pages are to be printed from the
same paper tray. The use of this setting is still confusing when there
is application-set tray for a page (e.g., for a page style in Writer,
through Format->Page), because the change in Preferences is applied (and
will be used for pages without application-set tray), but Preferences
shows the application-set value on second try again...

IOW, it still s---s, but it s---s a bit less .-)
---
 vcl/source/gdi/print3.cxx |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/vcl/source/gdi/print3.cxx b/vcl/source/gdi/print3.cxx
index 4824203..817e6d5 100644
--- a/vcl/source/gdi/print3.cxx
+++ b/vcl/source/gdi/print3.cxx
@@ -773,7 +773,7 @@ PrinterController::PageSize vcl::ImplPrinterControllerData::modifyJobSetup( cons
     PrinterController::PageSize aPageSize;
     aPageSize.aSize = mpPrinter->GetPaperSize();
     awt::Size aSetSize, aIsSize;
-    sal_Int32 nPaperBin = mnDefaultPaperBin;
+    sal_Int32 nPaperBin = (mnFixedPaperBin != -1) ? mnFixedPaperBin : mnDefaultPaperBin;
     for( sal_Int32 nProperty = 0, nPropertyCount = i_rProps.getLength(); nProperty < nPropertyCount; ++nProperty )
     {
         if( i_rProps[ nProperty ].Name.equalsAsciiL( RTL_CONSTASCII_STRINGPARAM( "PreferredPageSize" ) ) )
-- 
1.7.7.6

